From ba5e39e096ee863f0716c8c0bb2a81128227f11b Mon Sep 17 00:00:00 2001
From: rleix <rayx.lei@intel.com>
Date: Thu, 14 Dec 2017 10:44:18 +0800
Subject: [PATCH 07/21] Add refresh all pages of UserGridView if the data is
 changed.

Sometimes the UserGridView is not refresh completely even the data is changed.
Currently it only called the notifyDataSetChanged() in PagerAdapter. But it is
not engough. It also need override the getItemPosition() to destory all items
and renew them while it is calling the notifyDataSetChanged() to refresh.

Test:
1. Click the overview button and tap on Owner.
2. Add new user by tapping on "Add user" icon.
3. Click the overview button and tap the "Owner" icon to switch user mode
from New user to Owner.
4. Click the overview button and check whether the "Owner", "New user",
"Add guest" and "Add user" icons on quicksetting container.

Change-Id: I658f0732c538cf3b17e0c9f4b1967a4a2d3df5e0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-53447
Signed-off-by: Lei,RayX <rayx.lei@intel.com>
---
 .../com/android/systemui/statusbar/car/UserGridView.java    | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridView.java b/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridView.java
index e551801..5dfe40f 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridView.java
@@ -355,10 +355,19 @@ public class UserGridView extends ViewPager {
         }
 
         @Override
+        public int getItemPosition(Object object) {
+            return POSITION_NONE;
+        }
+
+        @Override
         public void onLayoutChange(View v, int left, int top, int right, int bottom,
                 int oldLeft, int oldTop, int oldRight, int oldBottom) {
-            mContainerWidth = Math.max(left - right, right - left);
-            notifyDataSetChanged();
+            int width = Math.max(left - right, right - left);
+            if (mContainerWidth != width) {
+                // Call notifyDataSetChanged() only the mContainerWidth is changed.
+                mContainerWidth = width;
+                notifyDataSetChanged();
+            }
         }
     }
 
-- 
1.9.1

